### PR TITLE
ARROW-6424: [C++] Fix IPC fuzzing test name

### PR DESCRIPTION
**Pull request by pitrou**

### CODE REVIEW

**Comment by pitrou** [comment]
@crepererum Can you check this fixes the issue for you?

File: cpp/src/arrow/filesystem/s3fs_test.cc [hunk 1]
```diff
@@ -166,8 +166,6 @@ Status MinioTestServer::Stop() {
 
 class S3Environment : public ::testing::Environment {
  public:
-  virtual ~S3Environment() {}
```
Comment (crepererum) [inline cpp/src/arrow/filesystem/s3fs_test.cc]: I think that change is unrelated?

File: cpp/cmake_modules/BuildUtils.cmake [hunk 2]
```diff
@@ -689,6 +689,9 @@ function(ADD_ARROW_FUZZING REL_FUZZING_NAME)
 
   get_filename_component(FUZZING_NAME ${REL_FUZZING_NAME} NAME_WE)
 
+  # Make sure the executable name contains only hyphens, not underscores
```
Comment (crepererum) [inline cpp/cmake_modules/BuildUtils.cmake]: that also does the job, thanks!

**Review by crepererum — CHANGES_REQUESTED** [review]

File: cpp/src/arrow/filesystem/s3fs_test.cc (hunk 1 above)
Comment (pitrou) [reply cpp/src/arrow/filesystem/s3fs_test.cc]: It is, but i had to make it to compile with clang.

Comment (crepererum) [reply cpp/src/arrow/filesystem/s3fs_test.cc]: Ah yes, we should probably include this then either here or in another PR.

**Review by crepererum — APPROVED** [review]
